hoomd/md/EvaluatorPairExpandedLJ.h

```diff
@@ -0,0 +1,227 @@
+// Copyright (c) 2009-2021 The Regents of the University of Michigan
+// This file is part of the HOOMD-blue project, released under the BSD 3-Clause License.
+
+// Maintainer: joaander
+
+#ifndef __PAIR_EVALUATOR_ExpandedLJ_H__
+#define __PAIR_EVALUATOR_ExpandedLJ_H__
+
+#ifndef __HIPCC__
+#include <string>
+#endif
+
+#include "EvaluatorPairExpandedLJ.h"
+#include "hoomd/HOOMDMath.h"
+
+/*! \file EvaluatorPairExpandedLJ.h
+    \brief Defines the pair evaluator class for Expanded LJ potentials
+*/
+
+// need to declare these class methods with __device__ qualifiers when building in nvcc
+// DEVICE is __host__ __device__ when included in nvcc and blank when included into the host
+// compiler
+#ifdef __HIPCC__
+#define DEVICE __device__
+#define HOSTDEVICE __host__ __device__
+#else
+#define DEVICE
+#define HOSTDEVICE
+#endif
+
+namespace hoomd
+    {
+namespace md
+    {
+//! Class for evaluating the Expanded LJ pair potential
+/*! <b>General Overview</b>
+
+    See EvaluatorPairExpandedLJ
+
+    <b>ExpandedLJ specifics</b>
+
+    EvaluatorPairExpandedLJ evaluates the function:
+    \f{eqnarray*}
+    V_{\mathrm{SLJ}}(r)  = & 4 \varepsilon \left[ \left( \frac{\sigma}{r - \Delta} \right)^{12} -
+                           \left( \frac{\sigma}{r - \Delta} \right)^{6} \right] & r <
+   (r_{\mathrm{cut}}) \\
+                         = & 0 & r \ge (r_{\mathrm{cut}}) \\
+    \f}
+
+    The ExpandedLJ potential does not need charge or diameter. Three parameters are specified and
+   stored in a Scalar3. \a lj1 is placed in \a params.x, \a lj2 is in \a params.y, and \a delta
+   is in params.z.
+
+
+    These are related to the standard lj parameters sigma and epsilon by:
+    - \a lj1 = 4.0 * epsilon * pow(sigma,12.0);
+    - \a lj2 = 4.0 * epsilon * pow(sigma,6.0);
+    - \a delta = delta
+
+    Due to the way that ExpandedLJ modifies the cutoff condition, it will not function properly with
+   the xplor shifting mode.
+*/
+class EvaluatorPairExpandedLJ
+    {
+    public:
+    //! Define the parameter type used by this pair potential evaluator
+    struct param_type
+        {
+        Scalar sigma_3;
+        Scalar sigma_6;
+        Scalar epsilon_x_4;
+        Scalar delta;
+
+        DEVICE void load_shared(char*& ptr, unsigned int& available_bytes) { }
+
+        HOSTDEVICE void allocate_shared(char*& ptr, unsigned int& available_bytes) const { }
+
+#ifdef ENABLE_HIP
+        //! Set CUDA memory hints
+        void set_memory_hint() const
+            {
+            // default implementation does nothing
+            }
+#endif
+
+#ifndef __HIPCC__
+        param_type() : sigma_6(0), epsilon_x_4(0), delta(0) { }
+
+        param_type(pybind11::dict v, bool managed = false)
+            {
+            auto sigma(v["sigma"].cast<Scalar>());
+            auto epsilon(v["epsilon"].cast<Scalar>());
+            delta = v["delta"].cast<Scalar>();
+
+            sigma_3 = sigma * sigma * sigma;
```

Comment:
```suggestion
            Scalar sigma_3 = sigma * sigma * sigma;
```
Make `sigma_3` a temporary variable.